pandas/tests/io/parser/test_header.py

```diff
@@ -61,6 +61,20 @@ def test_negative_multi_index_header(all_parsers, header):
         parser.read_csv(StringIO(data), header=header)
 
 
+@pytest.mark.parametrize("header", [([0, 0]), ([1, 0])])
+def test_nonincreasing_multi_index_header(all_parsers, header):
```

Comment:
I might be missing something, but the ``[0, 0]`` case seems to work 

```
    1   2   3   4   5
    1   2   3   4   5
0   6   7   8   9  10
1  11  12  13  14  15
```

Is there a reason why we don't try to make it work instead of raising an error? I get the code complexity argument, but I haven't looked if this would really add complexity.

Also we should probably deprecate first. I would consider this a bug on our side right now, raising is an unexpected change in behavior.

Reply:
@phofl Thanks for taking a look at my PR.

I suspect that the complexity of handling decreasing header arguments is beyond my current pandas skills. The current code handles all the permutations of named/unnamed Index/MultiIndex on both axes in both the Python and C parsers, and I'd probably break it while trying to consume header rows out-of-sequence. Since the user can .swaplevel today, and since the current code fails silently on decreasing header arguments, I think raising a ValueError is an improvement. A more clever person may still implement decreasing header arguments in the future, being careful to replace `header[-1]` with `max(header)` in the current code.

On the other hand, since header=[n, n] does work maybe we should keep it. I can't imagine a use case for it, but that could be a lack of creativity on my part. Would you prefer we keep it so we don't have to log a deprecation and follow up later?

Reply:
Since passing the same object twice works right now (also gives the correct result), we would have to deprecate before removing. Can't think of an example either, but this does not mean that it is not out there.

Decreasing headers work already for the python engine, so we would only have to fix the c engine. So I am -1 on raising

Reply:
The c engine fix is equally simple, so I would propose just fixing it.

Reply:
Excellent! I'll close this PR and attempt the fix. Thanks for the investigation/encouragement.